Makefile

```diff
@@ -164,10 +190,36 @@ stop-etcd:
 
 # Make sure that a copyright statement exists on all go files.
 check-copyright:
-	./check-copyrights.sh 
+	./check-copyrights.sh
+
+
+###############################################################################
+# push images of any tag
+###############################################################################
+imagetag:
+ifndef IMAGETAG
+	$(error IMAGETAG is undefined - run using make <target> IMAGETAG=X.Y.Z)
+endif
+
+push: image imagetag
```

Comment:
I think I'd rather not have the `push` target include `image` as a dependency, especially if it is going to be used as part of `make release`.

Reply:
I don't mind removing it, but since `image` just creates the image, which uses the cache, and we have a flag file indicating it has been done, why not?